syntax = "proto3";

message Ved {
	/*
	References:
	 - https://dfir.blog/google-ved-versions/
	 - https://deedpolloffice.com/blog/articles/decoding-ved-parameter
	 - http://gqs-decoder.blogspot.com/2013/08/google-referrer-query-strings-debunked-part-1.html
	*/

	int32 link_index = 1; // i (link index)
	int32 link_type = 2; // t (link type)
	int32 v3 = 3;
	int32 v4 = 4;
	int32 sub_result_position = 5; // (zero based)
	int32 result_position = 6; // r (result position)
	int32 results_start = 7; // s (start result position)
	int32 v8 = 8;
	int32 v9 = 9;
	int32 v10 = 10;
	int32 v11 = 11;
	int32 v12 = 12;
	message v13_Outer_Msg {
		message v13_Inner_Msg {
			int64 timestamp = 1;
			fixed32 v13_2 = 2;
			fixed32 v13_3 = 3;
		}
		v13_Inner_Msg v13_Inner = 1;
	}
	v13_Outer_Msg v13_Outer = 13;
	int32 v14 = 14;  // Haven't seen this one yet, not sure if used or if type is correct
	message v15_Msg {
	    int32 v15_1 = 1;
	    int32 v15_2 = 2;
	}
	v15_Msg v15 = 15;
}
